ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessConfigSubmission.scala

```diff
@@ -41,136 +41,147 @@ private[kvutils] case class ProcessConfigSubmission(
     )
   }
 
-  private def checkTtl(): Commit[Unit] = delay {
+  private val checkTtl: Commit[Unit] = delay {
     // Check the maximum record time against the record time of the commit.
     // This mechanism allows the submitter to detect lost submissions and retry
     // with a submitter controlled rate.
     val maxRecordTime = parseTimestamp(configSubmission.getMaximumRecordTime)
     if (recordTime > maxRecordTime) {
       logger.warn(
         s"Rejected configuration submission. The submission timed out ($recordTime > $maxRecordTime)")
-      rejectTimedOut
+      reject(
+        _.setTimedOut(
+          DamlConfigurationRejectionEntry.TimedOut.newBuilder
+            .setMaximumRecordTime(configSubmission.getMaximumRecordTime)
+        ))
     } else {
       pass
     }
   }
 
-  private def authorizeSubmission(): Commit[Unit] = delay {
+  private val authorizeSubmission: Commit[Unit] = delay {
     // Submission is authorized when:
-    //   1) The authorized participant is unset
-    //   2) The authorized participant matches the submitting participant.
+    //      the provided participant id matches source participant id
+    //  AND (
+    //      there exists no current configuration
+    //   OR the current configuration's participant matches the submitting participant.
+    //  )
+    val submittingParticipantId = configSubmission.getParticipantId
+    val wellFormed = participantId == submittingParticipantId
+
     val authorized =
-      currentConfig.authorizedParticipantId
-        .fold(true)(authPid => authPid == participantId)
+      currentConfigEntry.fold(true) {
```

Comment:
There's a much shorter and imho clearer way to write this: `currentConfigEntry.forall(_.getParticipantId == participantId)`